packages/shared/src/sdk-versioning/index.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  SDKConnectionInterface,
+  SDKLanguage,
+} from "back-end/types/sdk-connection";
+import uniq from "lodash/uniq";
+import { SDKCapability } from "./types";
+
+import * as javascript_json from "./sdk-versions/javascript.json";
+import * as nodejs_json from "./sdk-versions/nodejs.json";
+import * as react_json from "./sdk-versions/react.json";
+import * as php_json from "./sdk-versions/php.json";
+import * as python_json from "./sdk-versions/python.json";
+import * as ruby_json from "./sdk-versions/ruby.json";
+import * as java_json from "./sdk-versions/java.json";
+import * as kotlin_json from "./sdk-versions/kotlin.json";
+import * as swift_json from "./sdk-versions/swift.json";
+import * as go_json from "./sdk-versions/go.json";
+import * as flutter_json from "./sdk-versions/flutter.json";
+import * as csharp_json from "./sdk-versions/csharp.json";
+import * as other_json from "./sdk-versions/other.json";
+
+type SDKRecords = Record<SDKLanguage, SDKData>;
+type SDKData = {
+  versions: SDKVersionData[];
+};
+type SDKVersionData = {
+  version: string;
+  capabilities?: string[];
+};
+
+const sdks: SDKRecords = {
+  javascript: javascript_json,
+  nodejs: nodejs_json,
+  react: react_json,
+  php: php_json,
+  python: python_json,
+  ruby: ruby_json,
+  java: java_json,
+  android: kotlin_json,
+  ios: swift_json,
+  go: go_json,
+  flutter: flutter_json,
+  csharp: csharp_json,
+  other: other_json,
+};
+
+const getSdkData = (language: SDKLanguage = "other"): SDKData => {
+  let sdkData: SDKData = sdks[language];
+  if (!sdkData) {
+    sdkData = sdks["other"];
+  }
+  return sdkData;
+};
+
+export const getCurrentSDKVersion = (
+  language: SDKLanguage = "other"
+): string => {
+  const sdkData = getSdkData(language);
+  const versions = sdkData?.versions || [];
+  const current = versions?.[0];
+  return current?.version || "0.0.0";
+};
+
+export const isSDKOutdated = (
+  language: SDKLanguage = "other",
+  version: string = "0.0.0"
+): boolean => {
+  const current = getCurrentSDKVersion(language);
+  return paddedVersionString(version) < paddedVersionString(current);
+};
+
+export const getSDKCapabilities = (
+  language: SDKLanguage = "other",
+  version: string = "0.0.0"
+): SDKCapability[] => {
+  language = language || "other";
+  version = version && version !== "0" ? version : "0.0.0";
+  const sdkData = getSdkData(language);
+  const versions = sdkData?.versions || [];
+  const matches = versions.filter(
+    (data) => paddedVersionString(data.version) <= paddedVersionString(version)
+  );
+  const capabilities = matches.reduce(
```

Comment:
Are capabilities per version additive to previous versions or are they an exhaustive list of all the capabilities for a given version? If they're exhaustive we could probably just take the capabilities from the matching version instead of reducing, but I'm guessing they're not exhaustive

Reply:
they are not exhaustive, they are incremental